imap_processing/cdf/global_attrs.py

```diff
@@ -30,24 +33,63 @@ class GlobalConstantAttrs:
     ----------
     GLOBAL_BASE:
         Global file attributes, including project, source_name, discipline, PI_name,
-        PI_affiliation, instrument_type, and mission_group. This should be the same
+        PI_affiliation, and mission_group. This should be the same
         for all instruments.
 
     """
 
-    GLOBAL_BASE: ClassVar[dict] = {
+    GLOBAL_BASE: Final[dict] = {
         "Project": "STP>Solar-Terrestrial Physics",
         "Source_name": "IMAP>Interstellar Mapping and Acceleration Probe",
         "Discipline": "Solar Physics>Heliospheric Physics",
-        "PI_name": "Dr. David J. McComas",
-        "PI_affiliation": [
-            "Princeton Plasma Physics Laboratory",
-            "100 Stellarator Road, Princeton, NJ 08540",
-        ],
-        "Instrument_type": "Particles (space)",
+        # TODO: CDF docs say this value should be "IMAP"
         "Mission_group": "IMAP>Interstellar Mapping and Acceleration Probe",
     }
 
+    pi_name: tuple[str] = ("Dr. David J. McComas",)
+    pi_affiliation: tuple[str, str] = (
+        "Princeton Plasma Physics Laboratory",
+        "100 Stellarator Road, Princeton, NJ 08540",
+    )
+
+    def output(
+        self,
+        pi_names: Optional[tuple] = None,
+        pi_affiliations: Optional[tuple] = None,
+    ):
+        """
+        Generate the output for the global attributes as a dictionary.
+
+        If pi_names or pi_affiliations are included, they are added to the output (in
+        addition to the global IMAP PI information.)
+
+        Parameters
+        ----------
+        pi_names : tuple
+            A tuple of names of instrument PIs to include in the global attributes of
+            the file. Can be any length
+        pi_affiliations : tuple
+            A tuple of affilitations for each PI. This should include univeristy
+            information. Can be any length
+
+        Returns
+        -------
+        dict
+            Global base attributes, plus pi_names and pi_attributes from the default
+            values and any provided keywords
+        """
+        output_name = list(self.pi_name + pi_names if pi_names else self.pi_name)
```

Comment:
Is the purpose of appending to PI names/affiliations defined in this file to add the instrument PI?

Reply:
Yes